x-pack/plugins/lens/public/indexpattern_datasource/dimension_panel/dimension_editor.tsx

```diff
@@ -106,14 +108,14 @@ export function DimensionEditor(props: DimensionEditorProps) {
     hideGrouping,
   } = props;
   const { fieldByOperation, operationWithoutField } = operationSupportMatrix;
-  const [
-    incompatibleSelectedOperationType,
-    setInvalidOperationType,
-  ] = useState<OperationType | null>(null);
 
   const selectedOperationDefinition =
     selectedColumn && operationDefinitionMap[selectedColumn.operationType];
 
+  const incompleteInfo = (state.layers[layerId].incompleteColumns ?? {})[columnId];
+  const incompleteOperation = (incompleteInfo?.operationType as OperationType) ?? null;
```

Comment:
nit: `as OperationType` should be unnecessary, `incompleteOperation` will be `string | null` without it as well.